src/main/java/de/tum/in/www1/artemis/service/ExerciseLifecycleService.java

```diff
@@ -0,0 +1,62 @@
+package de.tum.in.www1.artemis.service;
+
+import java.time.ZonedDateTime;
+import java.util.concurrent.ScheduledFuture;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.enumeration.ExerciseLifecycle;
+
+@Service
+public class ExerciseLifecycleService {
+
+    private final Logger log = LoggerFactory.getLogger(ExerciseLifecycleService.class);
+
+    private TaskScheduler scheduler;
+
+    ExerciseLifecycleService(@Qualifier("taskScheduler") TaskScheduler scheduler) {
```

Comment:
Is there a reason that this is not public
I guess all other services are public as well even if we do not want them to be instantiated